ngraph_bridge/pass/transpose_sinking.cc

```diff
@@ -79,38 +78,41 @@ static string describe(shared_ptr<ngraph::Node> node) {
 }
 
 static shared_ptr<opset::Transpose> make_transpose(
-    shared_ptr<ngraph::Node> arg, const ngraph::AxisVector& input_order) {
-  auto ng_input_order = std::make_shared<opset::Constant>(
+    ngraph::Output<ngraph::Node> arg, const ngraph::AxisVector& input_order) {
+  auto order = std::make_shared<opset::Constant>(
       ngraph::element::u64, ngraph::Shape{input_order.size()}, input_order);
-  auto transpose = make_shared<opset::Transpose>(arg, ng_input_order);
+  auto transpose = make_shared<opset::Transpose>(arg, order);
   NGRAPH_VLOG(4) << "Make Transpose " << describe<opset::Transpose>(transpose);
   return transpose;
 }
 
 static shared_ptr<opset::Reshape> make_reshape(
-    shared_ptr<ngraph::Node> arg, const ngraph::AxisVector& input_order) {
-  auto ng_input_order = std::make_shared<opset::Constant>(
+    ngraph::Output<ngraph::Node> arg, const ngraph::AxisVector& input_order) {
+  auto order = std::make_shared<opset::Constant>(
       ngraph::element::u64, ngraph::Shape{input_order.size()}, input_order);
-  auto transpose = make_shared<opset::Reshape>(arg, ng_input_order, false);
+  auto transpose = make_shared<opset::Reshape>(arg, order, false);
   NGRAPH_VLOG(4) << "Make Reshape " << describe<opset::Reshape>(transpose);
   return transpose;
 }
 
 static void write_transposemap(TransposeMap& reorders,
-                               shared_ptr<ngraph::Node> target,
+                               ngraph::Output<ngraph::Node> target,
                                shared_ptr<opset::Transpose> transpose) {
-  NGRAPH_VLOG(4) << "Write TransposeMap[" << target->get_name()
+  auto name =
+      target.get_node()->get_name() + "." + to_string(target.get_index());
+  NGRAPH_VLOG(4) << "Write TransposeMap[" << name
                  << "] = " << describe<opset::Transpose>(transpose);
-  reorders[target].push_back(transpose);
+  reorders[name] = transpose;
 }
 
 static shared_ptr<opset::Transpose> read_transposemap(
     TransposeMap& reorders, ngraph::Output<ngraph::Node> target) {
-  auto transpose = reorders.at(target.get_node_shared_ptr());
-  NGRAPH_VLOG(4) << "Read TransposeMap[" << target.get_node()->get_name()
-                 << "]  -> "
-                 << describe<opset::Transpose>(transpose[target.get_index()]);
-  return transpose[target.get_index()];
+  auto name =
+      target.get_node()->get_name() + "." + to_string(target.get_index());
```

Comment:
This would be an issue if the same output from the node goes to two different nodes

Reply:
Not sure I understand. Can you elaborate?

Reply:
Node n 
                              /\
                             /. \
                   Dst x     Dst y
In this case the key in the map for output x would be n.0 and for y would also be n.0, right?

Reply:
No? The index is considered when generating the name.

Reply:
Yes, and name is the key so when writing in the map this could be a problem

Reply:
Ok, so `n` in the example above is not a multi-output op but a single-output op whose inputs are fed to more than one user?

Reply:
Yes, n is a single output op whose output is fed to more than 1 user

Reply:
Based on offline discussion, looks ok for now. We can look for a corner case where this might be an issue.